crates/flux-syntax/src/surface.rs

```diff
@@ -507,6 +513,7 @@ pub enum ExprKind {
     Alias(AliasReft, Vec<Expr>),
     IfThenElse(Box<[Expr; 3]>),
     Constructor(Option<ExprPath>, Vec<ConstructorArg>),
+    BoundedQuant(QuantKind, RefineParam, usize, usize, Box<Expr>),
```

Comment:
Let's use `std::ops::Range<usize>` for all these to clarify the intent. It may be slightly annoying because `Range<usize>` is not copy, so you'll have to insert some `.clone()`s but I think it'll be worth it.

```suggestion
    BoundedQuant(QuantKind, RefineParam, Range<usize>, Box<Expr>),
```

Reply:
hmmm, then let's use `core::range::Range`. You'll have to add `#[feature(new_range_api)]`

Reply:
grr. `core::range::Range` does not implement `Encodable` ... ?

Reply:
oh god, then disregard this comment...